src/components/ExpandableGridRow.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from 'react';
+import { DataView } from '../types';
+import { TezosDataView } from './TezosDataView';
+
+interface ExpandableGridRowProps {
+    dataView: DataView;
+    network: string;
+    data: Object;
+    rowNumber: number;
+}
+
+interface ExpandableGridRowState {
+    isExpanded: boolean;
+    expandedData: object;
+}
+
+export class ExpandableGridRow extends React.Component<ExpandableGridRowProps, ExpandableGridRowState> {
+
+    constructor(props: ExpandableGridRowProps) {
+        super(props);
+        this.state = {isExpanded: false, expandedData: {}};
+        this.handleOnClick = this.handleOnClick.bind(this);
+    }
+
+    handleOnClick(event: React.MouseEvent<HTMLButtonElement>) {
+        this.setState({isExpanded: !this.state.isExpanded});
+    }
+
+    inferID(data: object, dataView: DataView) {
+        const hash = 'hash';
+        const accountId = 'accountId';
+        if (dataView === DataView.Blocks || dataView === DataView.Operations) {
+            return data[hash];
+        } else if (dataView === DataView.Accounts) {
+            return data[accountId];
+        }
+    }
+
+    public render() {
+        if (!this.state.isExpanded) {
+            return (
+                <tr key={'row' + this.props.rowNumber}>
+                    <td><button onClick={this.handleOnClick}>+</button></td>
+                    {Object.keys(this.props.data).map((value, index) =>
```

Comment:
It looks like you just want the value of the object and not they key itself since you are taking the keys and then just using `this.props.data[value]`.

Instead, you could do `Object.values(this.props.data)` - this returns the values of the object rather than the keys